Common/Data/Model/File.swift

```diff
@@ -119,34 +187,36 @@ extension File {
 // MARK: computed properties
 extension File {
 
-    public var url: URL {
-        get {
-            return URL(string: currentPath)!
-        }
-        set {
-            self.currentPath = newValue.absoluteString
-        }
+    // TODO: replace with fileprovidermanager when implemented
+    static var localContainerURL: URL {
+        return FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: "group.org.schulcloud")!.appendingPathComponent("File Provider Storage")
     }
 
-    public var cacheUrl: URL? {
-        get {
-            guard let urlString = cacheURL_ else { return nil }
-            return URL(string: urlString)!
-        }
-        set {
-            cacheURL_ = newValue?.absoluteString
-        }
+    public var fileLocation: URL? {
+        guard self.downloadState == .downloaded else { return nil }
+        return self.localURL
+    }
+
+    public var localURL: URL {
+        let allowedCharacters = CharacterSet.whitespacesAndNewlines.inverted
+        return File.localContainerURL.appendingPathComponent("\(self.id)__\(self.name.addingPercentEncoding(withAllowedCharacters: allowedCharacters)!)")
```

Comment:
please extract this to a separate helper method like `self.localFileName` for `"\(self.id)__\(self.name)"`